docs/pubsub-api.rst

```diff
@@ -0,0 +1,255 @@
+``gcloud.pubsub`` API
+=====================
+
+Connection / Authorization
+--------------------------
+
+- Inferred defaults used to create connection if none configured explicitly:
+
+  - credentials (derived from GAE / GCE environ if present).
+
+  - ``project_id`` (derived from GAE / GCE environ if present).
+
+  - ``scopes``
+
+
+Manage topics for a project
+---------------------------
+
+Create a new topic for the default project:
+
+.. doctest::
+
+   >>> from gcloud import pubsub
+   >>> topic = pubsub.create_topic("topic_name")
+   >>> topic.name
+   'topic_name'
+
+Create a new topic for an explicit project:
+
+.. doctest::
+
+   >>> from gcloud import pubsub
+   >>> topic = pubsub.create_topic("topic_name", project_id="my.project")
+   >>> topic.name
+   'topic_name'
+
+Fetch an extant topic for the default project:
+
+.. doctest::
+
+   >>> from gcloud import pubsub
+   >>> topic = pubsub.get_topic('topic_name')
+   >>> topic.name
+   'topic_name'
+
+Fetch an extant topic for the default project:
+
+.. doctest::
+
+   >>> from gcloud import pubsub
+   >>> topic = pubsub.get_topic('topic_name', project_id="my.project")
+   >>> topic.name
+   'topic_name'
+
+Attempt to fetch a non-extant topic:
+
+.. doctest::
+
+   >>> from gcloud import pubsub
+   >>> pubsub.get_topic('nonesuch')
+   Traceback (most recent call last):
+     File "<stdin>", line 1, in ?
+   NotFound: ...
+
+List extant topics for the default project:
+
+.. doctest::
+
+   >>> from gcloud import pubsub
+   >>> [topic.name for topic in pubsub.list_topics()]
+   ['topic_name']
+
+List extant topics for an explicit project:
+
+.. doctest::
+
+   >>> from gcloud import pubsub
+   >>> [topic.name for topic in pubsub.list_topics(project_id="my.project")]
+   ['topic_name']
+
+Delete a topic:
+
+.. doctest::
+
+   >>> from gcloud import pubsub
+   >>> topic = pubsub.get_topic('topic_name')
+   >>> topic.delete()
```

Comment:
Should we offer a `pubsub.delete_topic('topic_name')` ? or just say to do `pubsub.get_topic('topic_name').delete()`

This would mean 2x the requests if you were deleting a lot of topics... Not sure how you guys feel about that.